src/state/slices/marketDataSlice/marketDataSlice.tsx

```diff
@@ -90,16 +76,22 @@ export const marketData = createSlice({
   initialState,
   reducers: {},
   extraReducers: builder => {
-    builder.addCase(fetchPriceHistory.pending, state => {
-      state.loading = true
+    builder.addCase(fetchPriceHistory.pending, (state, { meta }) => {
+      const { asset, timeframe } = meta.arg
+      state.priceHistory[timeframe][asset].loading = true
     })
-    builder.addCase(fetchPriceHistory.rejected, state => {
-      state.loading = false
+    builder.addCase(fetchPriceHistory.rejected, (state, { meta }) => {
+      const { asset, timeframe } = meta.arg
```

Comment:
we keep hitting rejected here even though we never see a network request go out - figure out why this is happening